src/Controls/tests/SourceGen.UnitTests/KnownTypeConvertersTests.cs

```diff
@@ -0,0 +1,309 @@
+using System.Globalization;
+using NUnit.Framework;
+using Microsoft.Maui.Controls.SourceGen;
+using Microsoft.Maui.Controls.Xaml;
+using Microsoft.CodeAnalysis;
+using System.IO;
+using System.Text;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using System.Linq;
+using Microsoft.Maui.Controls.Xaml.UnitTests.SourceGen;
+
+namespace Microsoft.Maui.Controls.SourceGen.UnitTests
+{
+	[TestFixture]
+	public class KnownTypeConvertersTests : SourceGenXamlInitializeComponentTestBase
+	{
+		[TestCase("en-US", "1.5*")]
+		[TestCase("fr-FR", "2.5*")]
+		[TestCase("de-DE", "3.14*")]
+		[TestCase("es-ES", "0.75*")]
+		[TestCase("ru-RU", "1.414*")]
+		public void GridLengthTypeConverter_StarValues_ProducesConsistentOutput_AcrossCultures(string cultureName, string gridLengthValue)
+		{
+			var originalCulture = CultureInfo.CurrentCulture;
+			var originalUICulture = CultureInfo.CurrentUICulture;
+			
+			try
+			{
+				// Set both current and UI cultures to test locale
+				CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo(cultureName);
+				CultureInfo.CurrentUICulture = CultureInfo.GetCultureInfo(cultureName);
```

Comment:
I'm never sure if changing these properties can affect other tests running in parallel or not. Do these change only the current thread culture? If it's just the current thread, does it propagate to the source generator (does it use other threads internally?). If it's a global proparty, maybe we should make these tests non-parallelizable?

Reply:
other tests are doing 
```
System.Threading.Thread.CurrentThread.CurrentCulture = ...;
```

Reply:
Made it non-parallelizable just in case, and using the API Stephane proposed.